compiler/aarch64/codegen/OMRTreeEvaluator.cpp

```diff
@@ -5624,10 +5624,358 @@ OMR::ARM64::TreeEvaluator::arraysetEvaluator(TR::Node *node, TR::CodeGenerator *
+    *    add     src2Reg, src2Reg, lengthReg
+    *    mov     lengthReg, #0
+    *    b       loop16
+    * } else {
+    *    add     lengthReg, lengthReg, #16
+    *    ;     arrayLen case
+    *    cbz     lengthReg, LDONE0
+    *    ;     non-arrayLen case
+    *    cbz     lengthReg, LDONE
+    *
+    *    b       lessThan16
+    * }
+    *
+    * Lnotequal16:                          ; src1Reg points 16-byte ahead of the location where data1reg was read.
+    * ;     arrayLen case
+    * eor     data3Reg, data3Reg, data4Reg
+    * cmp     data1Reg, #0
+    * csel    data1Reg, data1Reg, data3Reg, ne
+    * mov     data2Reg, #1
+    * cinc    data2Reg, data2Reg, ne
+    * sub     src1Reg, src1Reg, data2Reg, lsl #3 ; Adjusts post incremented address
+    * rbit    data1Reg, data1Reg
+    * clz     data1Reg, data1Reg
+    * add     src1Reg, src1Reg, data1Reg, lsr #3
+    *
+    * ;     non-arrayLen case
+    * cmp     data1Reg, data2Reg
+    * csel    data1Reg, data1Reg, data3Reg, ne
+    * csel    data2Reg, data2Reg, data4Reg, ne
+    * rev64   data1Reg, data1Reg
+    * rev64   data2Reg, data2Reg
+    *
+    * if !(length is constant and length > 15) {
+    *    b       LDONE0
+    *
+    *    LessThan16:
+    *    byteloop:
+    *    subs    lengthReg, lengthReg, #1
+    *    ldrb    data1Reg, [src1Reg], #1
+    *    ldrb    data2Reg, [src2Reg], #1
+    *    ccmp    data1Reg, data2Reg, #0, hi
+    *    b.eq    byteloop
+    *    ;     arrayLen case
+    *    cmp     data1Reg, data2Reg
+    *    cset    offReg, ne
+    *    sub     src1Reg, src1Reg, offReg
+    * }
+    * ;     arrayLen case
+    * LDONE0:
+    * sub     resultReg, src1Reg, savedSrc1Reg
+    * ;     non-arrayLen case
+    * LDONE0:
+    * cmp     data1Reg, data2Reg
+    * cset    resultReg, ne
+    * cinc    resultReg, resultReg, hi     ; Returns 1 or 2
+    *
+    * LDONE:
+    */
+   TR::Node *src1Node = node->getFirstChild();
+   TR::Node *src2Node = node->getSecondChild();
+   TR::Node *lengthNode = node->getThirdChild();
+   bool isLengthGreaterThan15 = lengthNode->getOpCode().isLoadConst() && lengthNode->getConstValue() > 15;
+   const bool isArrayCmpLen = node->isArrayCmpLen();
+   TR_ARM64ScratchRegisterManager *srm = cg->generateScratchRegisterManager(12);
+
+   TR::Register *savedSrc1Reg = cg->evaluate(src1Node);
+   TR::Register *src1Reg;
+   if ((src1Node->getReferenceCount() > 1) || isArrayCmpLen)
+      {
+      src1Reg = srm->findOrCreateScratchRegister();
+      generateMovInstruction(cg, node, src1Reg, savedSrc1Reg);
+      }
+   else
+      {
+      src1Reg = savedSrc1Reg;
+      }
+   TR::Register *src2Reg = cg->gprClobberEvaluate(src2Node);
+   TR::Register *lengthReg = cg->gprClobberEvaluate(lengthNode);
+   TR::Register *resultReg = cg->allocateRegister();
+   TR::LabelSymbol *startLabel = generateLabelSymbol(cg);
+   TR::LabelSymbol *doneLabel = generateLabelSymbol(cg);
+   TR_Debug * debugObj = cg->getDebug();
```

Comment:
I would suggest changing the style to `TR_Debug * debugObj = ...`: no space between the `*` and the variable name.

Reply:
Removed the space.